File: server/core/chromatic.py
from core_py import LocalPlayer, WaveGenerator, Note
import core_py

p = LocalPlayer()

wg = WaveGenerator()

note_length = core_py.sample_rate//8
note_intensity = core_py.max_intensity//4
print(type(note_length),type(note_intensity))
notes = [Note(i, -5, note_intensity, note_length*i, note_length*(i+1)) for i in range(140)]

for note in notes:
    wg.add_note(note)

p.add_instrument(wg)
p.play()

File: server/core/demos/chromatic.cpp
#include <cmath>
#include <iostream>
#include <stdexcept>
#include "../Player.h"
#include "../Instrument.h"
#include "../WaveGenerator.h"
using std::cout;
using std::endl;

int main(int argc, char ** argv)
{
    std::shared_ptr<Player> p(new LocalPlayer());
    InstrumentHandle wg = InstrumentHandle(new WaveGenerator());
    std::vector<Note> notes;

    guint64 note_length = Config::sample_rate/8;
    Sample note_intensity = Config::max_intensity/4;

    for(int i = 0; i < 140; ++i)
    {
        notes.push_back(Note(i,-5,note_intensity,note_length*i,note_length*(i+1)));
    }

    wg->add_notes(notes);
    p->add_instrument(wg);
    p->play();
}
